internal/resources/ekscluster/resource_ekscluster.go

```diff
@@ -316,27 +325,211 @@ func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m
+		!reflect.DeepEqual(meta.Labels, tmcCluster.Meta.Labels) {
+		updateCluster = true
+		tmcCluster.Meta.Description = meta.Description
+		tmcCluster.Meta.Labels = meta.Labels
+	}
+
+	if !reflect.DeepEqual(clusterSpec, tmcCluster.Spec) {
+		updateCluster = true
+	}
+
+	// The TF update request was only for nodepools.
+	// No need to update Cluster, it will error out.
+	if !updateCluster {
+		return nil
+	}
+
+	// there is some translation error, which results
+	// in mismatch on the server.
+	tmcCluster.Meta.CreationTime = strfmt.DateTime{}
 
-	_, err = config.TMCConnection.EKSClusterResourceService.EksClusterResourceServiceUpdate(
+	newCluster := &eksmodel.VmwareTanzuManageV1alpha1EksclusterEksCluster{
+		FullName: tmcCluster.FullName,
+		Meta:     tmcCluster.Meta,
+		Spec:     clusterSpec,
+	}
+
+	_, err := config.TMCConnection.EKSClusterResourceService.EksClusterResourceServiceUpdate(
 		&eksmodel.VmwareTanzuManageV1alpha1EksclusterCreateUpdateEksClusterRequest{
-			EksCluster: getResp.EksCluster,
+			EksCluster: newCluster,
 		},
 	)
 
 	if err != nil {
-		return diag.FromErr(errors.Wrapf(err, "Unable to update Tanzu Mission Control EKS cluster entry, name : %s", d.Get(NameKey)))
+		return errors.Wrapf(err, "Unable to update Tanzu Mission Control EKS cluster entry, name : %s", tmcCluster.FullName.Name)
 	}
 
-	log.Printf("[INFO] cluster update successful")
+	return nil
+}
 
-	return dataSourceTMCEKSClusterRead(ctx, d, m)
+func handleNodepoolDiffs(config authctx.TanzuContext, opsRetryTimeout time.Duration, clusterFn *eksmodel.VmwareTanzuManageV1alpha1EksclusterFullName, nodepools []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolDefinition) error {
+	npresp, err := config.TMCConnection.EKSNodePoolResourceService.EksNodePoolResourceServiceList(clusterFn)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list nodepools for cluster: %s", clusterFn)
+	}
+
+	npPosMap := nodepoolPosMap(nodepools)
+	tmcNps := map[string]*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolNodepool{}
+
+	npUpdate := []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolDefinition{}
+	npCreate := []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolDefinition{}
+	npDelete := []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolFullName{}
+
+	for _, tmcNp := range npresp.Nodepools {
+		tmcNps[tmcNp.FullName.Name] = tmcNp
+
+		if pos, ok := npPosMap[tmcNp.FullName.Name]; ok {
+			// np exisits in both TMC and TF
+			newNp := nodepools[pos]
+			fillTMCSetValues(tmcNp.Spec, newNp.Spec)
+
+			if checkNodepoolUpdate(tmcNp, newNp) {
+				fmt.Printf("np %s diff: %s", tmcNp.FullName.Name, cmp.Diff(tmcNp.Spec, newNp.Spec))
+				npUpdate = append(npUpdate, newNp)
+			}
+		} else {
+			// np exisits in TMC but not in TF
+			npDelete = append(npDelete, tmcNp.FullName)
+		}
+	}
+
+	for _, tfNp := range nodepools {
+		if _, ok := tmcNps[tfNp.Info.Name]; !ok {
+			npCreate = append(npCreate, tfNp)
+		}
+	}
+
+	for _, np := range npCreate {
```

Comment:
suggestion, since we have dedicated function to handle create, update and delete of an nodepool. 
can the same function be enchanced to take the array and perform the required action